osu.Game/Screens/Edit/Components/TimeInfoContainer.cs

```diff
@@ -36,6 +51,8 @@ protected override void Update()
         {
             base.Update();
             trackTimer.Text = editorClock.CurrentTime.ToEditorFormattedString();
+            bpm.Text = @$"{editorBeatmap.ControlPointInfo.TimingPointAt(editorClock.CurrentTime).BPM:0} BPM";
+            bpm.X = 5 - trackTimer.DrawWidth;
```

Comment:
The layouting in this component is kind of... wacky, and it's kind of visible:

![image](https://user-images.githubusercontent.com/20418176/170329172-9e73b93c-4f4f-4f0f-a887-5f5ccda215c2.png)

- Acres of space on the left of the clock portion, for no discernible reason (design has less)
- Not sure why the clock uses `CentreRight` anchoring while the BPM display uses `TopLeft`
- BPM label is just ever so slightly misaligned even despite this `bpm.X` set

Is there a reason this is like this, or was this simply left for a follow-up? A comment is removed in this diff that is talking about "going negative", but I don't think the clock *can* actually go negative in the lazer editor...?

Reply:
BPM is using `TopLeft` as it is aligned with the number text. I guess we could make it a fixed location, as if time goes negative it may get too far offset as it stands.

The empty space to the left was a concern, but in the design it was used by indicator lights. I'll look to remove it for now as it's a simple change.

Time going negative may happen in the future if it doesn't already, to allow for storyboard intros and the likes.

Reply:
The main point I wanted to make is that *as long as* negative time is not a concern, then the simplest possible way to align the texts is just to give them `X = 0` and a `TopLeft` anchor. This may get complicated if you want the negative time "minus" to always take up space even if time is positive, to avoid things shifting around when moving from negative time to positive.

> Time going negative may happen in the future if it doesn't already, to allow for storyboard intros and the likes.

As far as I can tell right now the clock display is hard-clamped to the track, and 00:00:000 is the start of the track, rather than first object (not sure how this should work, but seems correct?)

Reply:
Left alignment works much better for now. I've updated to do that.